[feat] Support `prepare_cmds` at the environment level (#2887)
reframe-hpc/reframe: docs/config_reference.rst

   .. versionadded:: 3.9.1

.. py:attribute:: environments.prepare_cmds

   :required: No
   :default: ``[]``

>    List of shell commands to be emitted after the :attr:`~config.systems.partitions.prepare_cmds` but before any module loading commands(if applicable) and environment variables are set.

Reviewer: I would not correlate this with the partition semantically. Whatever is defined in the environment should be scoped to it. So better rephrase it as:

```suggestion
   List of shell commands to be emitted before any commands that load the environment.
```
